source/firstprogram.rst

```diff
@@ -1,402 +1,621 @@
 
-The Network class gives a sense of geography to the model. The different components of the Network are simple locations like houses, offices, schools etc. The agents move about between these locations as shown in the figure below.
+Creating an Empty Class:
+^^^^^^^^^^^^^^^^^^^^^^^^
+Before we get to the actual coding aspect of the code, it is essential that we set up an empty and make sure that the framework is not giving any errors. 
+Navigate to ``BharatSim/src/main/scala/com/bharatsim/examples/epidemiology`` in IntelliJ or VSCode and right click on the folder. Upon right clicking, select the New Package option and name it SIR_Scratch. An empty folder should appear named SIR_Scratch and create a new scala class called Main. A yellow circular icon should appear. 
 
-The first step to building your model is to create locations which are part of the Network, by creating a new scala `class <https://docs.scala-lang.org/tour/classes.htm>`_. You can create a new class for adding houses to the network, in the following way:-
+The empty class should look like this. 
 
-1. Since ``House`` is a component of the Network, you have to import the Network class.
+.. image:: _static/images/New_class.png
+
+Now we can define a main function that has no input and has no output. The syntax and indentation of defining a function is as follows
 
 .. code-block:: scala
 
-  import com.bharatsim.engine.models.Network
+    def main(args: Array[String]): Unit = {
+    }
+
+The args means that the argument or the input is an array of Strings and the output is of type Unit, which corresponds to void means that there is no output. The code should look like this, 
 
-2. The ``House`` class is a `case class <https://docs.scala-lang.org/tour/case-classes.html>`_ and it extends the framework defined Network class.
+.. image:: _static/images/EmptyClass.png
+
+On running this, the output message should read ``Process finished with exit code 0``
+
+Inputting a File :
+^^^^^^^^^^^^^^^^^^
+
+To begin we must import a series of libraries and the function of each libraries will be explained as and when they are required. 
 
 .. code-block:: scala
 
-  case class House extends Network
+  import com.bharatsim.engine.Context
+  import com.bharatsim.engine.ContextBuilder._
+  import com.bharatsim.engine.execution.Simulation
+  import com.bharatsim.engine.graph.ingestion.{GraphData, Relation}
+  import com.typesafe.scalalogging.LazyLogging
+  import com.bharatsim.engine.utils.Probability.biasedCoinToss
+  import com.bharatsim.engine.basicConversions.encoders.DefaultEncoders._
 
-3. However, each house in the Network requires a unique house id which is given as an argument to the ``House`` class. This house id is an ‘attribute’ of the corresponding house.
+There needs to be a modification in the line where we have defined the object. We need to make use of a keywork called ``extends`` which allows one class to inherit the properties of another class. 

```

Comment:
First show the new line (i.e., show the line with the `extends LazyLogging`) and then explain why it's there, i.e. say something like:

You will notice that the line where we defined the `object` now has an `extends LazyLogging` in it. This is done [...]